new funcs: export CSV, undo, redo, pagination

novotnikg opened the pull request:
Download survey data as csv
Undo/redo in survey and cave editor
Pagination in survey and cave editor

novotnikg committed:
new funcs: export CSV, undo, redo, pagination

joemeszaros commented on src/ui/editor.js:
```diff
@@ -1425,7 +1425,15 @@ class SurveyEditor extends Editor {
+      {
+        id    : 'export-to-csv',
+        text  : 'Export to CSV',
+        click : () => this.table.download('csv', this.survey.name + '.csv', { delimiter: ';' })
```
please add the name of the cave to the CSV file's name in addition to the survey name

joemeszaros commented on src/ui/editor.js:
```diff
@@ -1515,6 +1524,12 @@ class SurveyEditor extends Editor {
       clipboardPasteAction  : 'range',
 
+      // pagination
+      pagination             : true,
```
please make this configurable via the config.js file. 
depending on the screen size one might find useful a configurable number of rows

joemeszaros reviewed (changes requested):
The downloaded file looks like this; 
```
"";"";"From";"To";"Length";"Azimuth";"Clino";"X";"Y";"Z";"Attributes"
"ok";"center";"0";"1";"5.1";"166.8";"-54.6";"644741.6746247347";"255548.12372276606";"-4.157151758215626";"[]"
```

The first 'status' field (with 'ok' values) should not be exported to the csv.
Please look at the following link: https://tabulator.info/docs/6.3/download#advanced-visibility
The second column does not have a name (splay type), pls figure it out why it's showing up as empty string.

novotnikg committed:
Pagination size from config.js
Download:
Add Cave name,
Remove status field,
Add missing column name: "Splay type"

novotnikg commented:
Pagination size from config.js

Download:
- Add Cave name,
- Remove status field,
- Add missing column name: "Splay type"
